src/js/components/Shared/TracksList.react.js

```diff
@@ -189,158 +124,240 @@ export default class TracksList extends Component {
-                            for(let i = 1; i <= Math.abs(index - onlySelectedIndex); i++) {
-                                selected.push(tracks[onlySelectedIndex - i]._id);
-                            }
-                        } else if(index > onlySelectedIndex) {
-                            for(let i = 1; i <= Math.abs(index - onlySelectedIndex); i++) {
-                                selected.push(tracks[onlySelectedIndex + i]._id);
-                            }
-                        }
-
-                        self.setState({ selected });
-                        break;
-                    }
-                    default: {
-                        const selectedInt = [];
-
-                        for(let i = 0, length = tracks.length; i < length; i++) {
-                            if(selected.includes(tracks[i]._id)) {
-                                selectedInt.push(i);
-                            }
-                        }
-
-                        let base;
-                        const min = Math.min(...selectedInt);
-                        const max = Math.max(...selectedInt);
-
-                        if(index < min) {
-                            base = max;
-                        } else {
-                            base = min;
-                        }
-
-                        const newSelected = [];
-
-                        if(index < min) {
-                            for(let i = 0; i <= Math.abs(index - base); i++) {
-                                newSelected.push(tracks[base - i]._id);
-                            }
-                        } else if(index > max) {
-                            for(let i = 0; i <= Math.abs(index - base); i++) {
-                                newSelected.push(tracks[base + i]._id);
-                            }
-                        }
-
-                        self.setState({ selected : newSelected });
-                        break;
-                    }
+    _buildTrackTiles() {
+        const self         = this,
+            selected       = this.state.selected,
+            tracks         = [...this.props.tracks],
+            trackPlayingId = this.props.trackPlayingId;
+
+        const chunkLength = 20;
+        const tilesToDisplay = 5;
+        const tileHeight = this.rowHeight * chunkLength;
+
+        const tracksChunked = utils.chunkArray(tracks, chunkLength);
+        const tilesScrolled = Math.floor(this.state.scrollTop / tileHeight);
+
+        return tracksChunked.splice(tilesScrolled, tilesToDisplay).map((tracksChunk, indexChunk) => {
+
+            const list = tracksChunk.map((track, index) => {
+
+                const trackRowIndex = (tilesScrolled + indexChunk) * chunkLength + index;
+
+                let playingIndicator = undefined;
+
+                if(trackPlayingId === track._id) {
+                    playingIndicator = <PlayingIndicator state={ this._pausePlayState() } />;
```

Comment:
Instead of deciding about rendering indicator at `TracksList` level, it might be better to decide it on
`TracksRow` level.

Example:
```js
const isCurrent = trackPlayingId === track._id;
const playingIndicator = <PlayingIndicator 
                            state={ this._pausePlayState() } 
                            isCurrent = { isCurrent } 
                         />;

// in TrackRow.react.js:
render() {
    if (!this.props.isCurrent) { return null; } // will render nothing

    // ... render if it IS current, just as it is now
}
```